jhbuild/modtypes/__init__.py

```diff
@@ -345,6 +346,129 @@ def _strip_debug_symbols(self, destdir_prefix, installroot):
             except Exception as e:
                 pass
 
+    def _find_exec(self, destdir_prefix):
+        exec_files = []
+        # filter out files to strip
+        for filename in fileutils.accumulate_dirtree_contents(destdir_prefix):
+            if os.access(os.path.join(destdir_prefix, filename), os.X_OK) or 'so' in os.path.basename(filename).split(os.path.extsep):
+                splited_filename = filename.split(os.path.sep)
+                # HACK from orininal bash script.
+                if 'scipy' in splited_filename:
+                    continue
+                if 'PIL' in splited_filename:
+                    continue
+                # Skip Qt5 libraries, which are already split and stripped
+                if 'libQt5' in os.path.basename(filename):
+                    continue
+                file_info = subprocess.check_output(['file', '-b', os.path.join(destdir_prefix, filename)])
+                # ignore symbolic link and stripped file
+                if 'ELF ' not in file_info:
+                    continue
+                exec_files.append(filename)
+
+        return exec_files
+
+    def _find_exec_ldd(self, destdir_prefix, filename):
+        filefullname = os.path.join(destdir_prefix, filename)
+        output = subprocess.check_output(['env', 'LD_LIBRARY_PATH=%s' % os.path.join(destdir_prefix, 'lib'), 'ldd', filefullname])
```

Comment:
use `env=` argument to Popen please. Also, please add the executable file's directory to LD_LIBRARY_PATH as well. That will get rid of a lot of the false alarms.